core/src/main/java/tachyon/conf/TachyonConf.java

```diff
@@ -0,0 +1,378 @@
+
+    InputStream defaultInputStream =
+        TachyonConf.class.getClassLoader().getResourceAsStream(DEFAULT_PROPERTIES);
+    if (defaultInputStream != null) {
+      try {
+        defaultProps.load(defaultInputStream);
+      } catch (IOException e) {
+        throw new RuntimeException("Unable to load default Tachyon properties file.");
+      }
+    }
+
+    // Update tachyon.master_address
+    String masterHostname = defaultProps.getProperty(Constants.MASTER_HOSTNAME);
+    String masterPort = defaultProps.getProperty(Constants.MASTER_PORT);
+    boolean useZk = Boolean.parseBoolean(defaultProps.getProperty(Constants.USE_ZOOKEEPER));
+    String masterAddress = (useZk ? Constants.HEADER_FT : Constants.HEADER) + masterHostname + ":"
+        + masterPort;
+    defaultProps.setProperty(Constants.MASTER_ADDRESS, masterAddress);
+
+    // Load site specific properties file
+    Properties siteProps = new Properties();
+    InputStream siteInputStream =
+        TachyonConf.class.getClassLoader().getResourceAsStream(SITE_PROPERTIES);
+    if (siteInputStream != null) {
+      try {
+        siteProps.load(siteInputStream);
+      } catch (IOException e) {
+        LOG.info("Unable to load site Tachyon configuration file.", e);
+      }
+    }
+
+    // Load system properties
+    Properties systemProps = new Properties();
+    if (includeSystemProperties) {
+      systemProps.putAll(System.getProperties());
+    }
+
+    // Now lets combine
+    mProperties.putAll(defaultProps);
+    mProperties.putAll(siteProps);
+    mProperties.putAll(systemProps);
+  }
+
+  /**
+   * @return the deep copy of the internal <code>Properties</code> of this TachyonConf instance.
+   */
+  public Properties getInternalProperties() {
+    return SerializationUtils.clone(mProperties);
+  }
+
+  /**
+   * Merge configuration properties with the other one. New one wins for duplicate
+   *
+   * @param alternateConf The source <code>TachyonConf</code> to be merged.
+   */
+  public void merge(TachyonConf alternateConf) {
+    if (alternateConf != null) {
+      // merge the system properties
+      mProperties.putAll(alternateConf.getInternalProperties());
+    }
+  }
+
+  // Public accessor methods
+
+  public void set(String key, String value) {
+    mProperties.put(key, value);
+  }
+
+  public String get(String key, final String defaultValue) {
+    String raw = mProperties.getProperty(key, defaultValue);
+    String updated = lookup(raw);
+    LOG.debug("Get Tachyon property {} as {} with default {}", key, updated, defaultValue);
+    return updated;
+  }
+
+  public int getInt(String key, final int defaultValue) {
+    if (mProperties.containsKey(key)) {
+      String rawValue = mProperties.getProperty(key);
+      try {
+        return Integer.parseInt(lookup(rawValue));
+      } catch (NumberFormatException e) {
+        LOG.warn("Configuration cannot evaluate key " + key + " as integer.");
```

Comment:
What's the convention for using {} replacement? In some parts of the code its used even if there is only 1 variable to be populated. Also, perhaps the value or the exception might also be helpful in this case?